tests/mocha/xml_test.js

```diff
@@ -73,13 +76,29 @@ suite('XML', function() {
   teardown(function() {
     sharedTestTeardown.call(this);
   });
+
   suite('textToDom', function() {
     test('Basic', function() {
       const dom = Blockly.utils.xml.textToDom(this.complexXmlText);
-      chai.assert.equal(dom.nodeName, 'xml', 'XML tag');
-      chai.assert.equal(dom.getElementsByTagName('block').length, 6, 'Block tags');
+      assertXmlDoc(dom);
+      chai.assert.equal(
+          dom.getElementsByTagName('block').length, 6, 'Block tags');
+    });
+
+    test('text with NCR Control characters are properly deserialized',
+        function() {
+          const dom = Blockly.utils.xml.textToDom('<xml>&#x1;</xml>');
+          assertXmlDoc(dom);
+          chai.assert.equal(dom.firstChild.textContent, ''); // u0001
```

Comment:
It looks like the character between the `''`s not a space but actually U+0001 / `0x01` / ascii `SOH` / `^a`—or rather it _completely fails to look like that_, the necessitating the comment.  Please use an escape sequence:

```suggestion
          chai.assert.equal(dom.firstChild.textContent, '\x01');
```
or
```suggestion
          chai.assert.equal(dom.firstChild.textContent, '\u0001');
```

(Despite the styleguide rule about comments for escape sequences I don't think one is required here because it's reasonably clear you are checking the decoding, rather than caring about the particular character in question.)

Reply:
(A type in the earlier version of the first suggestion has been fixed.)